developer-hub/src/reference_docs/defensive_programming.rs

```diff
@@ -0,0 +1,558 @@
+//! As our runtime should _never_ panic; this includes carefully handling [`Result`]/[`Option`]
+//! types, eliminating the possibility of integer overflows, converting between number types, or
+//! even handling floating point usage with fixed point arithmetic to mitigate issues that come with
+//! floating point calculations.
+//!
+//! Intentional and predictable design should be our first and foremost
+//! property for ensuring a well running, safely designed system.
+//!
+//! ## Defensive Programming
+//!
+//! Defensive programming is a design paradigm that enables a particular program to continue
+//! running despite unexpected behavior. These unforseen circumstances may
+//! cause the program to stop, (or in the Rust context, `panic!`) defensive practices allow for
+//! these circumstances to be accounted for ahead of time, and for them to be handled in a more
+//! graceful manner.
+//!
+//! The Polkadot SDK is both built to reflect these principles, and to be used accordingly.
+//!
+//! ## General Practices
+//!
+//! When developing within the context of the a runtime, there is one golden rule:
+//!
+//! ***DO NOT PANIC,*** most of the time. There are some exceptions, such as critical operations being actually more
+//! dangerous than allowing the node to continue running (block authoring, consensus, etc).
+//!
+//!  General guidelines:
+//!
+//! - Avoid panickable operations, such as directly using `unwrap()` for a [`Result`], common errors such as accessing collections out of bounds (using safer methods to access collection types, i.e., `get()`).
+//! - It may be acceptable to use `except()`, but only if one is completely certain (and has
+//!   performed a check beforehand) that a value won't panic upon unwrapping.
+//! - If you are writing a function that could panic, [be sure to document it!](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components)
```

Comment:
I think we should not write such functions in general.  
However, there are some. And normally we suffix them with `_unchecked` to make it obvious, since most devs dont read the docs 😆